tests/e2e/test_rosacli_cluster.go

```diff
@@ -854,6 +855,126 @@ var _ = Describe("Classic cluster creation validation",
+					for additionalSecurityGroupFlag, value := range securityGroups {
+						rosalCommand.AddFlags("--dry-run",
+							additionalSecurityGroupFlag, value,
+							"--version", "4.15.11",
+							"--hosted-cp", "-y")
+						output, err := rosaClient.Runner.RunCMD(strings.Split(rosalCommand.GetFullCommand(), " "))
+						Expect(err).To(HaveOccurred())
+						index = strings.Index(additionalSecurityGroupFlag, "a")
+						flagName = additionalSecurityGroupFlag[index:]
+						Expect(output.String()).To(ContainSubstring("Using '301721915996' as billing account"))
+						Expect(output.String()).To(ContainSubstring("add --billing-account xxxxxxxxxx to previous command"))
+						Expect(output.String()).To(ContainSubstring("Hosted clusters deprecate the --multi-az flag"))
+						Expect(output.String()).To(ContainSubstring(
+							"Parameter '%s' is not supported for Hosted Control Plane clusters",
+							flagName))
+						rosalCommand.DeleteFlag(additionalSecurityGroupFlag, true)
+					}
+					Skip("Security groups not supported from Hosted Control Plane cluster")
+				}
+
+				if profile.ClusterConfig.BYOVPC {
+					By("Try creating cluster with additional security groups and ocp version lower than 4.14")
+					for additionalSecurityGroupFlag, value := range securityGroups {
+						levelMap := map[string]string{
+							additionalSecurityGroupFlag: value,
+							"--version":                 ocpVersionBelow4_14,
+						}
+
+						rosalCommand.AddFlags(
+							"--dry-run",
+							additionalSecurityGroupFlag, levelMap[additionalSecurityGroupFlag],
+							"--version", ocpVersionBelow4_14, "-y")
+						output, err := rosaClient.Runner.RunCMD(strings.Split(rosalCommand.GetFullCommand(), " "))
+						Expect(err).To(HaveOccurred())
+						index = strings.Index(additionalSecurityGroupFlag, "a")
+						flagName = additionalSecurityGroupFlag[index:]
+						Expect(output.String()).To(ContainSubstring(
+							"Parameter '%s' is not supported prior to version '4.14.0'",
+							flagName))
+						rosalCommand.DeleteFlag(additionalSecurityGroupFlag, true)
+					}
+
+					By("Try creating cluster with invalid additional security groups")
+					for additionalSecurityGroupFlag, value := range invalidSecurityGroups {
+						rosalCommand.AddFlags("--dry-run", additionalSecurityGroupFlag, "invalid", "--version", ocpVersion4_14, "-y")
+						output, err := rosaClient.Runner.RunCMD(strings.Split(rosalCommand.GetFullCommand(), " "))
+						Expect(err).To(HaveOccurred())
+						Expect(output.String()).To(ContainSubstring("Creating cluster"))
+						Expect(output.String()).To(ContainSubstring(
+							"To view a list of clusters and their status, run 'rosa list clusters'"))
+						Expect(output.String()).To(ContainSubstring("Security Group ID '%s' doesn't have 'sg-' prefix", value))
+						rosalCommand.DeleteFlag(additionalSecurityGroupFlag, true)
+					}
+
+					By("Try creating cluster with additional security groups with invalid and more than 5 SG ids")
```

Comment:
I assume we're supposed to fail cluster creation if there are more than 5 Security Groups provided, but I'm not sure that this actually covers that? If we're trying to confirm that we can't input more than 5 security groups, wouldn't it make more sense to use valid IDs here to ensure the error is with the number of SGs rather than the SG itself?

Reply:
@OAharoni-RedHat i agree with your thoughts, i guess `invalid` word should be replaced with `valid` here at this test step `Create cluster with invalid  --additional-infra-security-group-ids ,--additional-control-plane-security-group-ids,  --additional-compute-security-group-ids with more than 5 SG ids` @xueli181114 could you please confirm this? Thanks!